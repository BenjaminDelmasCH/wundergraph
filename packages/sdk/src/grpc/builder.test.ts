import * as fs from 'fs';
import GrpcSchemaBuilder from './builder';
import { printSchema } from 'graphql';

const runTest = async (testFile: string, snapShot: string) => {
	const exists = fs.existsSync(testFile);
	if (!exists) {
		return;
	}
	const fileContents = fs.readFileSync(testFile);

	const gprc = new GrpcSchemaBuilder(fileContents, '');
	const Schema = await gprc.Schema();

	const printedSchema = printSchema(Schema);
	expect(printedSchema).toMatchSnapshot(snapShot + '_' + 'schema');

	// fs.writeFileSync('src/grpc/' + snapShot + '.grapqhl', printedSchema);
};

test('starwars grpc', async () => {
	await runTest('../../pkg/datasources/grpc/testdata/starwars/starwars.protoset', 'starwars_grpc');
});
