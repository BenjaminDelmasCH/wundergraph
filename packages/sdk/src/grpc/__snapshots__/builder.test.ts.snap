// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starwars grpc: starwars_grpc_schema 1`] = `
"type Query {
  starwars_StarwarsService_GetHero(input: starwars_GetHeroRequest_Input): starwars_Character
  starwars_StarwarsService_GetHuman(input: starwars_GetHumanRequest_Input): starwars_Character
  starwars_StarwarsService_GetDroid(input: starwars_GetDroidRequest_Input): starwars_Character
  starwars_StarwarsService_ListHumans(input: starwars_ListEmptyRequest_Input): starwars_ListHumansResponse
  starwars_StarwarsService_ListDroids(input: starwars_ListEmptyRequest_Input): starwars_ListDroidsResponse
}

type starwars_Character {
  id: String
  name: String
  friends: [starwars_Character]
  appears_in: [starwars_Episode]
  home_planet: String
  primary_function: String
  type: starwars_Type
}

enum starwars_Episode {
  _
  NEWHOPE
  EMPIRE
  JEDI
}

enum starwars_Type {
  HUMAN
  DROID
}

input starwars_GetHeroRequest_Input {
  episode: starwars_Episode
}

input starwars_GetHumanRequest_Input {
  id: String
}

input starwars_GetDroidRequest_Input {
  id: String
}

type starwars_ListHumansResponse {
  humans: [starwars_Character]
}

scalar starwars_ListEmptyRequest_Input @specifiedBy(url: \\"http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf\\")

type starwars_ListDroidsResponse {
  droids: [starwars_Character]
}"
`;
